src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -635,8 +635,11 @@ public void run()
          final List<PoolBagEntry> bag = connectionBag.values(STATE_NOT_IN_USE);
          int removable = bag.size() - config.getMinimumIdle();
          for (PoolBagEntry bagEntry : bag) {
+            if (removable <= 0) {
+               break;
+            }
             if (connectionBag.reserve(bagEntry)) {
-               if (removable > 0 && idleTimeout > 0L && clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout) {
+               if (idleTimeout > 0L && clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout) {
                   closeConnection(bagEntry, "(connection passed idleTimeout)");
```

Comment:
Ew, that just looks icky.

Admitedly less efficient ... bypassing the reservation but still not breaking the loop.  Likely less than a microsecond of "spin" even if the pool had 500 idle connections.  Reads much nicer:
```java
for (PoolBagEntry bagEntry : bag) {
   if (removable > 0L && connectionBag.reserve(bagEntry)) {
      ...
   }
}
```